packages/data-store-web/src/entities/document/create-document.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  ContextProps,
+  DataStoreEvents,
+  WalletDocument,
+} from '@docknetwork/wallet-sdk-data-store/src/types';
+import {v4 as uuid} from 'uuid';
+import {DocumentEntity} from './document.entity';
+import {getDocumentById} from './get-document-by-id';
+import {logger} from '@docknetwork/wallet-sdk-data-store/src/logger';
+import {localStorageJSON} from '../../localStorageJSON';
+import {getAllDocuments} from './get-all-documents';
+
+export async function appendDocument(document: WalletDocument) {
+  const entities = (await localStorageJSON.getItem('documents')) || [];
+
+  await localStorageJSON.setItem('documents', [...entities, document]);
+}
+
+/**
+ * Create document
+ * @param dataStore
+ * @param json
+ */
+export async function createDocument({
+  dataStore,
+  json,
+  options,
+}: ContextProps & {
+  json: any;
+  options?: any;
+}): Promise<WalletDocument> {
+  logger.debug(`Creating document with id ${json.id}...`);
+  if (json.id) {
+    const existingDocument = await getDocumentById({
+      dataStore,
+      id: json.id,
+    });
+
+    if (existingDocument) {
+      logger.debug(`Document with id ${json.id} already exists`);
+      throw new Error(`Document with id ${json.id} already exists`);
+    }
+  }
+
+  if (!json.id) {
```

Comment:
what happens if the id is specified but a document with that id already exists? Does it overwrite the old one?

Reply:
throwing a document with id ${json.id} already exists